tests/integration/test_schema_registry_auth.py

```diff
@@ -0,0 +1,213 @@
+"""
+karapace - schema registry authentication and authorization tests
+
+Copyright (c) 2022 Aiven Ltd
+See LICENSE for details
+"""
+from karapace.client import Client
+from karapace.kafka_rest_apis import KafkaRestAdminClient
+from karapace.schema_models import SchemaType, ValidatedTypedSchema
+from tests.utils import (
+    new_random_name,
+    new_topic,
+    schema_avro_json,
+    schema_jsonschema_json,
+    test_objects_avro,
+    wait_for_topics,
+)
+from typing import List
+from urllib.parse import quote
+
+import aiohttp
+import asyncio
+import requests
+
+NEW_TOPIC_TIMEOUT = 10
+
+admin = aiohttp.BasicAuth("admin", "admin")
+aladdin = aiohttp.BasicAuth("aladdin", "opensesame")
+reader = aiohttp.BasicAuth("reader", "secret")
+
+
+async def test_sr_auth(registry_async_client_auth: Client) -> None:
+    subject = new_random_name("cave-")
+
+    res = await registry_async_client_auth.post(f"subjects/{quote(subject)}/versions", json={"schema": schema_avro_json})
+    assert res.status_code == 401
+
+    res = await registry_async_client_auth.post(
+        f"subjects/{quote(subject)}/versions", json={"schema": schema_avro_json}, auth=aladdin
+    )
+    assert res.status_code == 200
+    sc_id = res.json()["id"]
+    assert sc_id >= 0
+
+    res = await registry_async_client_auth.get(f"subjects/{quote(subject)}/versions/latest")
+    assert res.status_code == 401
+    res = await registry_async_client_auth.get(f"subjects/{quote(subject)}/versions/latest", auth=aladdin)
+    assert res.status_code == 200
+    assert sc_id == res.json()["id"]
+    assert ValidatedTypedSchema.parse(SchemaType.AVRO, schema_avro_json) == ValidatedTypedSchema.parse(
+        SchemaType.AVRO, res.json()["schema"]
+    )
+
+
+# Test each endpoint without authorization to verify those require authentícation
+async def test_sr_auth_endpoints(registry_async_client_auth: Client) -> None:
+    subject = new_random_name("any-")
+
+    res = await registry_async_client_auth.post(
+        f"compatibility/subjects/{quote(subject)}/versions/1", json={"schema": schema_avro_json}
+    )
+    assert res.status_code == 401
+
+    res = await registry_async_client_auth.get(f"config/{quote(subject)}")
+    assert res.status_code == 401
+
+    res = await registry_async_client_auth.put(f"config/{quote(subject)}", json={"compatibility": "NONE"})
+    assert res.status_code == 401
+
+    res = await registry_async_client_auth.get("config")
+    assert res.status_code == 401
+
+    res = await registry_async_client_auth.put("config", json={"compatibility": "NONE"})
+    assert res.status_code == 401
+
+    res = await registry_async_client_auth.get("schemas/ids/1/versions")
+    assert res.status_code == 401
+
+    res = await registry_async_client_auth.get("schemas/types")
+    assert res.status_code == 200
+
+    res = await registry_async_client_auth.get("schemas/types", auth=admin)
+    assert res.status_code == 200
```

Comment:
Why test with `admin`? Also, the test explanation is not exactly correct, it says endpoints require authentication, but `schemas/types` doesn't need that. Maybe split testing that to another function. That way this test function would be consistently testing only the endpoints requiring authentication.